### PR TITLE
silence compiler warnings

### PR DESCRIPTION
**Pull request by vailkyte**

### CODE REVIEW

**Comment by sonninnos** [comment]
Please no, this is pointless.. Only the libretro part can be accepted, but touching the VICE codebase is a waste of time, since it gets replaced periodically. Rather silence the warnings with compilation flags.

**Comment by vailkyte** [comment]
This worked with windows mingw builds
```unsigned timer = strcmp(var.value, "On") == 0 ? 150 : 0;```
(It would toggle an osd debug message on and off)

Recent build of linux mingw always print debug messages and never toggles off anymore.

Some other compiler mentioned it was getting confused by the ternary parsing order.

When I did this
```unsigned timer = (strcmp(var.value, "On") == 0) ? 150 : 0;```

it works okay again.

There's lots of places in the VICE code that has this "problem" - sometimes it's there, other times not. libretro-common also plenty of these ambiguous ternary orders.

This parsing behavior broke some core emulation of SameBoy. I directly mentioned it there but it was treated as non-compiler error.

`==============`
`==============`

tl;dr
It's okay - I'll shut this down and use it in my own private builds. I have given up convincing developers it's a serious problem worth () guarding.

**Comment by sonninnos** [comment]
Like I said, convincing me does no good since this is not the actual VICE repo. The fixes are needed in upstream, and then they trickle here.

Any related libretro-side fixes should be done of course, like I also said.

**Comment by sonninnos** [comment]
Interesting why I got a mail from sourceforge as if I had sent something to VICE devs from my email even though I did no such thing.. Any explanations?

>
>Your mail to 'Vice-emu-mail' with the subject
>
>   ternary compiler bug
>
>Is being held until the list moderator can review it for approval.
>
>The reason it is being held:
>
>   Post by non-member to a members-only list

**Comment by mrdudz** [comment]
I can confirm at least that i got that mail through the vice ml :)

We are always trying to fix all warnings - please make a bug report on the sourceforge tracker (and perhaps include the diffs). It would also be interesting which compiler exactly produces those warnings - and using which options. I have never seen them for that matter (gcc 12.3.1 on my box).

**Comment by sonninnos** [comment]
I haven't seen any such warnings and issues with any platforms I've tried with though.

**Comment by mrdudz** [comment]
It must be either a bleeding edge compiler - or some insane settings. We've generally enabled a lot of warnings in the VICE build system, so i am wondering as well.

**Comment by dqh-au** [comment]
How is

unsigned timer = strcmp(var.value, "On") == 0 ? 150 : 0;

Ambiguous? '==' has a higher precedence than '?:'. This is a very common pattern
